### PR TITLE
Load critical css and rest asynchronously

### LINKED ISSUE #7891: Load only critical css for intial page load and rest asynchronously
Currently the application.css file is blocking the rendering of the website. But most of the css present in it is unused for the initial page load or not critical. So it makes sense to defer this unused css and only load the critical css for the initial page load.

**Please describe the desired behavior.**
The loadcss-rails [gem](https://github.com/michael-misshore/loadcss-rails) and the critical-path-css-rails [gem](https://github.com/mudbugmedia/critical-path-css-rails) can be used for the same.

loadcss-rails gem helps to load all the css asynchronously by installing and making changes in the layouts where we are calling the application.css file such as application.html.erb layout.

critical-path-css-rails gem can help to only load the critical css for the inital page load. This is done by installing and configuring in the same layout. We can see it in action by precompiling our assets.

**Additional context (optional)**
Coverage tab showing us how much of application.css is unused.  

image

Page SpeedInsight showing us this as the suggestion-    

image

### PR DESCRIPTION
**Pull request by Tlazypanda**
Fixes #7891 

Load only critical css and rest asynchronously for better performance. The loadcss-rails gem and the critical-path-css-rails gem is used for the same.

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

Thanks!

### CODE REVIEW

**Comment by Tlazypanda** [comment]
@jywarren Can you kindly review? I have also pushed this on unstable to test it out :sweat_smile:

File: config/critical_path_css.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  # css_paths:
+  #   - /path/to/css/from/public/main.css
+  routes:
+    - /
+
+development:
+  <<: *defaults
+  base_url: http://localhost:3000
+
+staging:
```
Comment (jywarren) [inline config/critical_path_css.yml:16]: Should we also make one for stable.publiclab.org?

File: lib/tasks/critical_path_css.rake [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require 'critical-path-css-rails'
+
+namespace :critical_path_css do
```
Comment (jywarren) [inline lib/tasks/critical_path_css.rake:3]: Awesome and way to test this out on unstable!!!

File: config/critical_path_css.yml (hunk 1 above)
Comment (jywarren) [reply config/critical_path_css.yml:16]: Perhaps we can just make this stable instead? Or would that mean it'd stop working in unstable?  Can we have both, or a wildcard like `*.publiclab.org`? Thank you!!!

Comment (Tlazypanda) [reply config/critical_path_css.yml:16]: The wildcard * is not working so I have modified it to stable :sweat_smile:

**Comment by Tlazypanda** [comment]
Hey @jywarren Updated the pr :v: can you kindly review again? Also for adding in the README.md I am not sure what section it should be a part of? I was thinking maybe we can create a separate doc on performance and link it to the readme where I can document all done till now to avoid future confusion :sweat_smile:

**Comment by jywarren** [comment]
trying to retrigger tests!

As to README, what about here? https://github.com/publiclab/plots2/blob/main/README.md#prerequisites

And just to confirm, this won't change any basic commands or introduce new required commands that we need to notify most developers of, right? Thank you!!!

**Comment by Tlazypanda** [comment]
Hey @jywarren actually I have started with this document on performance and accessibility here https://hackmd.io/@itm2017004/BJuqtap6L maybe I can add it here? :sweat_smile: and later I can add a section in https://github.com/publiclab/plots2/blob/main/README.md#prerequisites linking this file? 

Also there are no new commands that need to be notified to others :sweat_smile: 
Travis is failing due to tag_selection_test thing so I think it is unrelated

**Comment by jywarren** [comment]
OK, gotcha! Thanks!

Perhaps we need to look into the underlying/unrelated inconsistent test. will drop some in here: #7315

**Comment by Tlazypanda** [comment]
Just looping in @VladimirMikulic I think he has solved this issue before as well :sweat_smile: @VladimirMikulic help us out!! :pray:

**Comment by Tlazypanda** [comment]
@jywarren build is passing now :sweat_smile:

**Comment by VladimirMikulic** [comment]
Sorry, @Tlazypanda for not coming earlier. By restarting Travis, I no longer have logs for the failing test so I can't troubleshoot it :(

**Comment by Tlazypanda** [comment]
@VladimirMikulic ohh sorry about that I think https://github.com/publiclab/plots2/pull/7948 is failing  and one of the failing test is tag selection test :sweat_smile: can you check this out instead? Thanks :v:

**Comment by jywarren** [comment]
Hi @Tlazypanda if you're able to rebase this that would be great, that way we see it built in GitPod, for a final check. Otherwise I think we can merge by end of day. Thank you very much!!!

**Comment by jywarren** [comment]
Ah, and just a note -- I think there's a conflict with the FontAwesome CDN link, that would need resolving in order to rebase.

**Comment by Tlazypanda** [comment]
Hey @jywarren I am so sorry I totally missed that this needs rebasing :sweat_smile: I have rebased now :v:

**Comment by jywarren** [comment]
Hi @Tlazypanda - i realized our latest code wasn't pushed to stable branch before publishing last week, so that is going live in an hour or two now that I've taken that step: https://jenkins.laboratoriopublico.org/job/Deploy%20Plots2/71/

Sorry for the mis-step!

**Pull request merged**